### PR TITLE
added JavaSpring beanValidation for RequestBody parameters fix #13932

### LINKED ISSUE #13932: [BUG] [SPRING] Missing Java Bean Validation annotations on RequestBody
##### Description

We are unable to generate beanValidation annotations for RequestBody params. Looks like the feature was missing in revision 8952815b5f7fef3f64de28afb95aa47392413872 

Proposed changes should cover this bug and allow beanValidation annotations to be generated on RequestBody params too.

##### openapi-generator version

6.2.1 , current master

##### OpenAPI declaration file content or url

Provided configuration:
```yaml
paths:
  /add:
    post:
      requestBody:
        content:
          application/json:
            schema:
              type: integer
              title: quantity
              minimum: 2
        required: true
      responses:
        '200':
          description: OK
          content:
            application/json:
              schema:
                type: boolean
```
Generates:
```java
@Operation(
        operationId = "addPost",
        responses = {
            @ApiResponse(responseCode = "200", description = "OK", content = @Content(mediaType = "application/json", schema = @Schema(implementation =  Boolean.class)))
        }
    )
    @RequestMapping(
        method = RequestMethod.POST,
        value = "/add",
        produces = { "application/json" },
        consumes = { "application/json" }
    )
    default ResponseEntity<Boolean> cartKekPost(
        @Parameter(name = "body", description = "", required = true) @Valid @RequestBody Integer body
    ) {
        return new ResponseEntity<>(HttpStatus.NOT_IMPLEMENTED);

    }
```
Must have:
```java


    default ResponseEntity<Boolean> addPost(
        @Min(2)  @Parameter(name = "body", description = "", required = true) @Valid @RequestBody Integer body
    ) {
        return new ResponseEntity<>(HttpStatus.NOT_IMPLEMENTED);

    }
```

##### Command line used for generation

Plugin configuration: (beanValidation is enabled by default)

generatorName = 'spring'
configOptions = [
			dateLibrary: 'java8',
			interfaceOnly: 'true',
			openApiNullable: 'false',
			java8: 'false'
]
importMappings = ['java.time.OffsetDateTime': 'java.time.Instant']
typeMappings = ['OffsetDateTime': 'Instant']

##### Steps to reproduce

./gradlew openApiGenerate

##### Suggest a fix/enhancement

will propose a solution as PR from fork repository https://github.com/Mintas/openapi-generator

### PR DESCRIPTION
**Pull request by Mintas**
fixes https://github.com/OpenAPITools/openapi-generator/issues/13932

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

mention JavaSpring community:
@cachescrubber (2022/02) @welshm (2022/02) @MelleD (2022/02) @atextor (2022/02) @manedev79 (2022/02) @javisst (2022/02) @borsch (2022/02) @banlevente (2022/02) @Zomzog (2022/09)

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/JavaSpring/bodyParams.mustache
Comment (borsch) [inline modules/openapi-generator/src/main/resources/JavaSpring/bodyParams.mustache:1]: probably would better to this `{{>beanValidationBodyParams}}` right after `@Valid`

**Review by borsch — CHANGES_REQUESTED** [review]

File: modules/openapi-generator/src/main/resources/JavaSpring/bodyParams.mustache
Comment (Mintas) [reply modules/openapi-generator/src/main/resources/JavaSpring/bodyParams.mustache:1]: agree, but made similar to queryParams. Worth swap?

Comment (borsch) [reply modules/openapi-generator/src/main/resources/JavaSpring/bodyParams.mustache:1]: as for me - yes, better to put after `@Valid`

Comment (Mintas) [reply modules/openapi-generator/src/main/resources/JavaSpring/bodyParams.mustache]: fixed

**Review by borsch — APPROVED** [review]

**Comment by Mintas** [comment]
@borsch what are my next steps to get this PR merged?

**Comment by borsch** [comment]
> @borsch what are my next steps to get this PR merged?

ask somebody with merge permissions to do so :)
https://github.com/OpenAPITools/openapi-generator#61---openapi-generator-core-team

**Comment by Mintas** [comment]
@borsch  @wing328  there are some failures, looks like the problem is in the CI pipeline itself (smth with JAVA_HOME and/or version)

**Comment by Mintas** [comment]
@wing328 lets merge this one?)

**Review by borsch — APPROVED** [review]

**Comment by borsch** [comment]
@Mintas could you please pull latest master into your branch and I'll merge once pipeline is successful

**Pull request merged**
